app/controllers/account/members_controller.rb

```diff
@@ -27,6 +27,32 @@ def destroy
     end
   end
 
+  def greeting_new_member
+    @invitation = Invitation.find_by(token: params[:token], workspace_id: params[:workspace_id])
+
+    if !@invitation
+      redirect_to root_path, notice: 'Sorry, could not identify following link'
+      return
+    end
+
+    if @invitation.expired?
+      redirect_to root_path, notice: 'Sorry, this link is no longer valid'
+    else
+      @workspace = Workspace.find(params[:workspace_id])
+      @invitor = User.find(@invitation.invitor.id)
+    end
+  end
+
+  def create_thought_link
+    @workspace = Workspace.find(params[:workspace_id])
```

Comment:
is it possible for any user to invite user to any workspace if I change URL directly?
what about security